### PR TITLE
[IOPLT-311] Adding integration tests to pipeline

### PR DESCRIPTION
**Pull request by drmarro**
Integration tests have been added to the project. 

#### List of Changes

The __integrations__folder contains all the tests and utilities.

#### Motivation and Context

To test the service endpoint with real dependencies.

#### Types of changes

- [X] Chore (nothing changes by a user perspective)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: __integrations__/utils/cosmos.ts [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  database: Database,
+): TE.TaskEither<Error, readonly Container[]> =>
+  pipe(
+    [
+      pipe(
+        TE.Do,
+        TE.bind("collection", () =>
+          createCollection(database, COSMOS_COLLECTION_NAME, "id"),
+        ),
+        TE.chainFirst(({ collection }) => upsertItem(collection, { id: ID })),
```
Comment (gunzip) [inline __integrations__/utils/cosmos.ts:56]:
```suggestion
        createCollection(database, COSMOS_COLLECTION_NAME, "id"),
        TE.chainFirst(( collection ) => upsertItem(collection, { id: ID })),
```

Comment (gunzip) [reply __integrations__/utils/cosmos.ts:56]: you can simplify avoiding bind ( I guess, haven't really tried : - )

File: __integrations__/utils/cosmos.ts [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+        RA.map((r) => deleteCollection(db, r)),
+        RA.sequence(TE.ApplicativePar),
+      ),
+    ),
+    TE.map((collections) => {
+      return collections;
+    }),
+    TE.mapLeft((err) => {
+      return err;
+    }),
```
Comment (gunzip) [inline __integrations__/utils/cosmos.ts:120]:
```suggestion
```

File: src/capturer/cosmos/cosmos.ts [hunk 3]
```diff
@@ -20,6 +20,13 @@ export const getChangeFeedIteratorOptions = (
   maxItemCount: maxItemCount && maxItemCount > 0 ? maxItemCount : 1,
 });
 
+// eslint-disable-next-line no-var
+var shouldExitExternal = false;
+
+export const setShouldExitExternal = (value: boolean): void => {
```
Comment (gunzip) [inline src/capturer/cosmos/cosmos.ts:26]: what's this for? I'd add a comment

File: src/capturer/cosmos/utils.ts [hunk 4]
```diff
@@ -11,12 +16,34 @@ export const ContinuationTokenItem = T.type({
...
   connectionString: string,
+): E.Either<Error, CosmosClientOptions> =>
+  pipe(
+    connectionString,
+    cosmosConnectionRegex,
+    O.fromNullable,
+    E.fromOption(
+      () => "cosmos connection string does not match the expected format",
+    ),
+    E.mapLeft((message) => new Error(message)),
```
Comment (gunzip) [inline src/capturer/cosmos/utils.ts:33]:
```suggestion
    E.mapLeft(E.toError),
```

File: src/factory/cosmosCDCService.ts [hunk 5]
```diff
@@ -9,45 +8,84 @@ import {
...
-          cosmosDBServiceClient.getResource(database, leaseResource),
+          pipe(
+            O.fromNullable(leaseResource),
+            O.fold(
+              () => TE.right<Error, O.Option<Container>>(O.none),
+              (lease) =>
+                pipe(
+                  cosmosDBServiceClient.getResource(database, lease),
+                  TE.map((lcontainer) => O.some(lcontainer)),
+                  TE.mapLeft((error) => error),
```
Comment (gunzip) [inline src/factory/cosmosCDCService.ts:47]:
```suggestion
```

**Review by gunzip — COMMENTED** [review]
It's very demanding to review such a long PR, next time I suggest to make several different PRs in order to let them be merged sooner.

File: .github/workflows/code-review.yaml [hunk 6]
```diff
@@ -43,4 +43,30 @@ jobs:
...
 
       - name: 'Code review'
-        run: yarn run code-review
\ No newline at end of file
+        run: yarn run code-review
+
+      - name: Set env
+        run: echo "NODE_TLS_REJECT_UNAUTHORIZED=0" >> $GITHUB_ENV
+
+      # - name: Start cosmosdb
```
[the remark this answers is not in the document]
Comment (drmarro) [reply .github/workflows/code-review.yaml:51]: This part is already in progress, there's a bug with the cosmosdb emulator running on linux containers on github actions and I'm trying to understand how to fix it

File: src/capturer/cosmos/cosmos.ts (hunk 3 above)
Comment (drmarro) [reply src/capturer/cosmos/cosmos.ts:26]: It's an external signal to exit from the loop that is waiting for database changes

**Pull request closed without merging**